### PR TITLE
Add flashpack to IDE

### PR DESCRIPTION
**Pull request by calvinatintel**
Needs to be paired with a json which installs flashpack.

### CODE REVIEW

**Comment by calvinatintel** [comment]
My [first attempt at the json](http://mkfs.ndg.intel.com/dev/json/package_0.1.53_index.json)

1. Installs the factory recovery flashpack
2. Populates **Programmer** entry
3. Triggers `flash_dfu.sh` upon clicking **Burn Bootloader** button

screen shot 2016-08-17 at 11 36 27 am

However, the flashpack hangs after the board is reset, and does not initiate flashing.
screen shot 2016-08-17 at 11 39 40 am

Considering that the board reappears on **Tools > Port** menu within 5 seconds, it seems that `flash_dfu.sh` script didn't catch the board in DFU mode. Running the script manually from within `.arduino15` directory correctly executed the script. Perhaps we need the baudrate reset mechanism built into the flashpack.

**Comment by calvinatintel** [comment]
A couple tasks left:

1. Reduce flashpack to BLE only. Maybe we can attach the `factory-ble-flashpack.tar.bz2` to Castor release and link it back to json? So meta.
2. Try to fix the hanging issue
    - Test on other OSes to see if this is an OSX quirk wrt quanrantine bit
    - Add the baudrate reset script into the flashpack
    - Make a wrapper script which calls the baudrate reset script, followed by `flash_dfu.sh`

**Comment by calvinatintel** [comment]
First attempt at [arduino101-factory_ble-flashpack.tar.bz2](http://mkfs.ndg.intel.com/jenkins_artifacts/pvt/arduino101-factory_ble-flashpack.tar.bz2).
Works on OSX. Need other OSes tested.

**Comment by eriknyquist** [comment]
Nice. will verify after kanban.

**Comment by bbaltz505** [comment]
Tested programmer on Windows. I had to add a Windows-specific instruction in platform.txt:
`tools.arduino101load.bootloader.pattern.windows=cmd /C {runtime.tools.flashpack.path}/flash_dfu.bat`
but it also seemed to hang

**Comment by noelpaz** [comment]
You verify this by using a BLE Central corelibs . we need to be testing the Central Examples. Scanning should scan for available BLE devices

**Comment by noelpaz** [comment]
Will test with Brian's fixes on windows and Ubuntu

File: platform.txt [hunk 1]
```diff
@@ -99,3 +99,11 @@ tools.arduino101load.upload.params.verbose=verbose
...
 
 tools.arduino101load.upload.pattern="{cmd.path}" "{runtime.tools.arduino101load.path}/x86/bin" {build.path}/{build.project_name}.bin {serial.port} "{upload.verbose}"
+
+# This is needed to avoid an error on unexistent fields
+tools.arduino101load.erase.params.verbose=
+tools.arduino101load.erase.params.quiet=
+tools.arduino101load.erase.pattern=
+tools.arduino101load.bootloader.params.verbose=
+tools.arduino101load.bootloader.params.quiet=
+tools.arduino101load.bootloader.pattern=/bin/bash {runtime.tools.flashpack.path}/flash_dfu.sh
```
Comment (calvinatintel) [inline platform.txt]: We need to make a version of this which works for both Linux, OSX, and Windows. Currently only works in Linux and OSX.

Comment (bigdinotech) [inline platform.txt]: If you look at how we did it in Edison you should see how we used a different method for each OS

Comment (bbaltz505) [inline platform.txt]: I pushed a fix for Windows. The other OSes should support bash.

Comment (calvinatintel) [inline platform.txt]: Awesome. Can you guys please test it?

File: boards.txt [hunk 2]
```diff
@@ -28,6 +28,9 @@ arduino_101.build.variant_system_lib=arc32drv_arduino101
 arduino_101.build.vid=0x8087
 arduino_101.build.pid=0x0AB6
 
+arduino_101.bootloader.tool=arduino101load
+arduino_101.bootloader.file=""
```
Comment (bbaltz505) [inline boards.txt]: This causes the IDE to give a warning:
`Bootloader file specified but missing: /home/babaltz/.arduino15/packages/Intel-Test/hardware/arc32/0.0.49/bootloaders/""`

Comment (calvinatintel) [inline boards.txt]: I saw that too. Since that file isn't used, perhaps we can point that to the actual `image.bin` to suppress the error?

Comment (bbaltz505) [inline boards.txt]: Removing that line seems to resolve the warning

Comment (calvinatintel) [inline boards.txt]: Alright, let's go with that

**Comment by calvinatintel** [comment]
I think we have a working solution. I'll publish a new json in 10 min.

**Comment by calvinatintel** [comment]
http://mkfs.ndg.intel.com/precommit/json/package_0.0.50_index.json

Works on OSX. Please verify on Linux and Windows.
If this works on all OSes, we're ready for 1.0.7.

**Comment by calvinatintel** [comment]
Verified on Linux 64 and Linux 32.

**Pull request merged**